### ISSUE TITLE
Cuda Error in executeMemcpy: 1 (invalid argument)

### ISSUE THREAD

**Issue by piaoliuping123** [nanmi/yolact-trt#2]
首先特别感谢您的yolact-trt开源，我按照流程步骤分别生成onnx，engine，但是在执行
context->enqueue(batch_size, (void**)buffers, stream, nullptr);
报错
[06/07/2022-15:39:31] [W] [TRT] TensorRT was linked against cuBLAS/cuBLAS LT 11.3.0 but loaded cuBLAS/cuBLAS LT 11.2.1
[06/07/2022-15:39:31] [W] [TRT] TensorRT was linked against cuBLAS/cuBLAS LT 11.3.0 but loaded cuBLAS/cuBLAS LT 11.2.1
engine.getNbBindings: 6
test1: 0 1 2 4 3
>>>>>>>>>>>>>>
[06/07/2022-15:39:31] [E] [TRT] ../rtSafe/cuda/genericReformat.cu (1294) - Cuda Error in executeMemcpy: 1 (invalid argument)
[06/07/2022-15:39:31] [E] [TRT] FAILED_EXECUTION: std::exception
>>>>>>>>>>>>>>
这个问题困扰我很久了，感谢你的答疑

**Comment by zzz3bbb3** [comment]
Thank you for your question. This problem seems to be caused by cuda memory operation. In addition, this problem [#1969](https://github.com/NVIDIA/TensorRT/issues/1969) occurs when I actually run it, but don't make a fuss, you can still get the correct result.

**Comment by piaoliuping123** [comment]
Thank you very much for your reply，I still have a doubt here that I can't get the correct test results.I did the following debug：

    // Create CPU buffers
    static float* h_input=nullptr, *loca=nullptr, *mask=nullptr, *cof=nullptr, *mmp=nullptr;

    Tbegin = std::chrono::steady_clock::now();

    CHECK(cudaMallocHost((void**)&h_input, i_size*sizeof(float)));
    memcpy(h_input, input_raw_data, i_size*sizeof(float));
    CHECK(cudaMallocHost((void**)&loca, loc_size*sizeof(float)));
    CHECK(cudaMallocHost((void**)&mask, mask_size*sizeof(float)));
    CHECK(cudaMallocHost((void**)&cof, cof_size*sizeof(float)));
    CHECK(cudaMallocHost((void**)&mmp, mask_map_size*sizeof(float)));

    CHECK(cudaMalloc((void**)&buffers[i_idx], i_size*sizeof(float)));
    CHECK(cudaMemcpyAsync(buffers[i_idx], h_input, i_size*sizeof(float), cudaMemcpyHostToDevice, stream));

    // cuda result
    CHECK(cudaMalloc((void**)&buffers[loc_idx], loc_size*sizeof(float)));
    CHECK(cudaMalloc((void**)&buffers[mask_idx], mask_size*sizeof(float)));
    CHECK(cudaMalloc((void**)&buffers[cof_idx], cof_size*sizeof(float)));
    CHECK(cudaMalloc((void**)&buffers[mask_map_idx], mask_map_size*sizeof(float)));
printf(">>>>>>>>>>>>>>\n");
    context->enqueue(batch_size, (void**)buffers, stream, nullptr);
printf(">>>>>>>>>>>>>>\n");

    CHECK(cudaMemcpyAsync(loca, buffers[loc_idx], loc_size*sizeof(float), cudaMemcpyDeviceToHost, stream));
    CHECK(cudaMemcpyAsync(mask, buffers[mask_idx], mask_size*sizeof(float), cudaMemcpyDeviceToHost, stream));
    CHECK(cudaMemcpyAsync(cof, buffers[cof_idx], cof_size*sizeof(float), cudaMemcpyDeviceToHost, stream));
    CHECK(cudaMemcpyAsync(mmp, buffers[mask_map_idx], mask_map_size*sizeof(float), cudaMemcpyDeviceToHost, stream));

    cudaStreamSynchronize(stream);
    Iend = std::chrono::steady_clock::now();
    float infer_time = std::chrono::duration_cast <std::chrono::milliseconds> (Iend - Tbegin).count();

    std::cout << "*loca: " << *loca << std::endl;
    std::cout << "*mask: " << *mask << std::endl;
    std::cout << "*cof: " << *cof << std::endl;
    std::cout << "*mmp: " << *mmp << std::endl;

and print *loca, *mask, *cof, *mmp: the values of these four variables, and the values of the four variables are all 0.
*loca: 0
*mask: 0
*cof: 0
*mmp: 0

please advise where I might be going wrong, thanks again


    const int i_idx = engine->getBindingIndex("input.1"); //1x3x550x550
    const int loc_idx= engine->getBindingIndex("776"); // 1x19248x4
    const int mask_idx = engine->getBindingIndex("778"); //1x19248x32
    const int cof_idx= engine->getBindingIndex("780"); // 1x19248x81
    const int mask_map_idx = engine->getBindingIndex("595"); // [1x32x138x138]

The .engine generated at the same time is a bit different from your getBindingIndex, so I am wondering if there is something wrong here, thank you.

**Comment by piaoliuping123** [comment]
I refer to this article https://zhuanlan.zhihu.com/p/510843962, 
but I get const int mask_map_idx = engine->getBindingIndex("595"); // [1x32x138x138], 
but I see your source The code is const int mask_map_idx = engine->getBindingIndex("588"); 
I look forward to your answer, thanks

**Comment by zzz3bbb3** [comment]
> 



> I refer to this article https://zhuanlan.zhihu.com/p/510843962, but I get const int mask_map_idx = engine->getBindingIndex("595"); // [1x32x138x138], but I see your source The code is const int mask_map_idx = engine->getBindingIndex("588"); I look forward to your answer, thanks

Yes, this article is also written by me. What I want to say is that the names of the input and output of different models may be slightly different. You need to set it according to the specific situation of the onnx model. Oh, I forgot to explain this.😀

**Comment by lixiangweiabc** [comment]
> Thank you very much for your reply，I still have a doubt here that I can't get the correct test results.I did the following debug：
> 
> ```
> // Create CPU buffers
> static float* h_input=nullptr, *loca=nullptr, *mask=nullptr, *cof=nullptr, *mmp=nullptr;
> 
> Tbegin = std::chrono::steady_clock::now();
> 
> CHECK(cudaMallocHost((void**)&h_input, i_size*sizeof(float)));
> memcpy(h_input, input_raw_data, i_size*sizeof(float));
> CHECK(cudaMallocHost((void**)&loca, loc_size*sizeof(float)));
> CHECK(cudaMallocHost((void**)&mask, mask_size*sizeof(float)));
> CHECK(cudaMallocHost((void**)&cof, cof_size*sizeof(float)));
> CHECK(cudaMallocHost((void**)&mmp, mask_map_size*sizeof(float)));
> 
> CHECK(cudaMalloc((void**)&buffers[i_idx], i_size*sizeof(float)));
> CHECK(cudaMemcpyAsync(buffers[i_idx], h_input, i_size*sizeof(float), cudaMemcpyHostToDevice, stream));
> 
> // cuda result
> CHECK(cudaMalloc((void**)&buffers[loc_idx], loc_size*sizeof(float)));
> CHECK(cudaMalloc((void**)&buffers[mask_idx], mask_size*sizeof(float)));
> CHECK(cudaMalloc((void**)&buffers[cof_idx], cof_size*sizeof(float)));
> CHECK(cudaMalloc((void**)&buffers[mask_map_idx], mask_map_size*sizeof(float)));
> ```
> 
> printf(">>>>>>>>>>>>>>\n"); context->enqueue(batch_size, (void**)buffers, stream, nullptr); printf(">>>>>>>>>>>>>>\n");
> 
> ```
> CHECK(cudaMemcpyAsync(loca, buffers[loc_idx], loc_size*sizeof(float), cudaMemcpyDeviceToHost, stream));
> CHECK(cudaMemcpyAsync(mask, buffers[mask_idx], mask_size*sizeof(float), cudaMemcpyDeviceToHost, stream));
> CHECK(cudaMemcpyAsync(cof, buffers[cof_idx], cof_size*sizeof(float), cudaMemcpyDeviceToHost, stream));
> CHECK(cudaMemcpyAsync(mmp, buffers[mask_map_idx], mask_map_size*sizeof(float), cudaMemcpyDeviceToHost, stream));
> 
> cudaStreamSynchronize(stream);
> Iend = std::chrono::steady_clock::now();
> float infer_time = std::chrono::duration_cast <std::chrono::milliseconds> (Iend - Tbegin).count();
> 
> std::cout << "*loca: " << *loca << std::endl;
> std::cout << "*mask: " << *mask << std::endl;
> std::cout << "*cof: " << *cof << std::endl;
> std::cout << "*mmp: " << *mmp << std::endl;
> ```
> 
> and print *loca, *mask, *cof, *mmp: the values of these four variables, and the values of the four variables are all 0. *loca: 0 *mask: 0 *cof: 0 *mmp: 0
> 
> please advise where I might be going wrong, thanks again
> 
> ```
> const int i_idx = engine->getBindingIndex("input.1"); //1x3x550x550
> const int loc_idx= engine->getBindingIndex("776"); // 1x19248x4
> const int mask_idx = engine->getBindingIndex("778"); //1x19248x32
> const int cof_idx= engine->getBindingIndex("780"); // 1x19248x81
> const int mask_map_idx = engine->getBindingIndex("595"); // [1x32x138x138]
> ```
> 
> The .engine generated at the same time is a bit different from your getBindingIndex, so I am wondering if there is something wrong here, thank you.

I met the same problem with yours, the values of the four variables are all 0. How did you solve it?
